virkailija/src/routes/HOKSLomake/helpers/koodiUriSelected.ts

```diff
@@ -1,25 +1,21 @@
 import { updateCodeVersionAt } from "./updateCodeVersionAt"
 
-export const koodiUriSelected = (self: any, callback?: () => void) => (
-  path: string,
-  selected: boolean
-) =>
+export const koodiUriSelected = (
+  setState: (state: any) => any,
+  callback?: () => void
+) => (path: string, selected: boolean) =>
   new Promise<void>(resolve => {
-    self.setState(
-      (state: any) => ({
-        ...state,
-        formData: updateCodeVersionAt(
-          path,
-          selected,
-          state.formData,
-          state.koodiUris
-        )
-      }),
-      () => {
```

Comment:
note to self: mitä tää setStaten toka argumentti oikein tekee

Reply:
Luokkakomponentin setState osasi ottaa callback-funktion, jota kutsuttiin kun state oli asetettu. Funktiokomponenteissa ei oo täysin vastaavaa, eli siksi siirsin callbackin kutsumisen setStaten kutsumisen jälkeen.